lib/model/CRuleScope.cc

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+#include <model/CRuleScope.h>
+
+#include <core/CPatternSet.h>
+
+#include <model/CAnomalyDetectorModel.h>
+#include <model/CDataGatherer.h>
+
+namespace ml {
+namespace model {
+
+void CRuleScope::include(std::string field, const core::CPatternSet& filter) {
+    m_Scope.emplace_back(field, TPatternSetCRef(filter), E_Include);
```

Comment:
Since you're passing `field` by value you should move it into `m_Scope`, i.e.:

```
    m_Scope.emplace_back(std::move(field), TPatternSetCRef(filter), E_Include);
```

Same in `exclude` immediately below.